kolibri/plugins/learn/assets/src/views/LibraryPage/AccessibleFolderCard.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+
+  <div>
+    <ul>
+      <KCard
+        class="card"
+        :to="to"
+        :headingLevel="2"
+        layout="horizontal"
+        thumbnailDisplay="large"
+        :title="contentNode.title"
+      >
+        <template #thumbnailPlaceholder>
+          <div>
+            <CardThumbnail
+              class="thumbnail"
+              :contentNode="contentNode"
+            />
+          </div>
+        </template>
+
+        <template #belowTitle>
+          <div class="">
+            <div
+              class="header-bar"
+              :style="headerStyles "
+            >
+              <div v-if="!contentNode.is_leaf">
+                <span class="inline-color">
+                <KIcon
+                  icon="topic"
+                  :color="$themeTokens.text"
+                  class="folder-header-bar"
+                />
+                <p class="folder-header-text">
+                  {{ coreString('folder') }}
+                </p>
+              </span>  
+              </div> 
+            </div>
+          </div>
+        </template>
+      </KCard>
+    </ul>
+  </div>
+
+</template>
+
+
+<script>
+
+  import { validateLinkObject } from 'kolibri.utils.validators';
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+  import useChannels from '../../composables/useChannels';
+  import commonLearnStrings from '../commonLearnStrings';
+  import CardThumbnail from './../HybridLearningContentCard/CardThumbnail.vue';
+
+  export default {
+    name: 'AccessibleFolderCard',
+    components: {
+      CardThumbnail, 
+    },
+    mixins: [commonLearnStrings, commonCoreStrings],
+    setup() {
```

Comment:
**[required]** Let's remove usage of `useChannels` from this component completely, and just receive `channelThumbnail` and `channelTitle` via props (you can move it to the page that contains the card)

For smaller components, this has long been a recommended pattern that detaches them from business logic (typically Vuex or nowadays some kinds of composables) and it increases flexibility and makes maintenance easier. It is also much easier to write tests because you don't need to use mocks, and simply feed the component with data via props and then see its output. It's a pattern that is sometimes called presentational (aka dumb, stateless) and container (aka business, smart, stateful) components. You can see https://www.patterns.dev/vue/container-presentational/.

Note that this pattern is undergoing some evolution with the emergence of composables, and may not apply in all cases, however I still think it's often relevant and in this case it will make the component simpler, better readable and testable.

Reply:
Ah actually now I noticed that we don't need them at all or am I missing something from future work? Please double check and if that's the case we don't even need the props for them I suggested. I am leaving the original comment for reference since it's a useful pattern.